aten/src/ATen/native/mps/operations/Copy.mm

```diff
@@ -178,15 +178,14 @@ void copy_cast_mps(at::Tensor& dst, const at::Tensor& src,
   id<MTLBuffer> destBuffer = getMTLBufferStorage(dst_);
   uint64_t src_total_size = 0;
 
-  // This is weird, but sometimes this function can be called
-  //  with contiguous destination and non-contiguous source
-  if (src_.is_view() || dst_.is_contiguous() != src_.is_contiguous()) {
+  // Get contiguous view if src and dst tensors are differently strided
+  if (src_.strides() != dst_.strides()) {
+    TORCH_CHECK(dst_.is_contiguous());
     src = src_.to(dst_.dtype()).expand_as(dst_).contiguous();
     // Get the actual size of a View (takes into account the storage offset)
     // For View tensors, the storage offset can be bigger than what's being reported by nbytes
     src_total_size = at::detail::computeStorageNbytesContiguous(src.sizes(), src.element_size(), src.storage_offset());
```

Comment:
I think this condition could be moved outside of these if check, so it's always consistent for view / contig tensors (computeStorageNbytesContiguous would work on both of them). 
For view tensors, `nbytes()` wouldn't report the whole size of the buffer, and when we do `newBufferWithBytesNoCopy`, we'd pass an incorrect size of the total memory. The current failing mps test is also related to this issue (likely it's a view tensors skipping the if check where the `src_total_size` is calculated).

Reply:
That's correct, I think earlier we were in-correctly going into the earlier  `if-check` for the cases where src and dst strides were same. Now with this change its going down the `else-case` which is where it should be but the size calculation is not correct.